xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -850,7 +847,7 @@ unsigned int CAESinkAUDIOTRACK::AddPackets(uint8_t **data, unsigned int frames,
           }
           else
           {
-            sleep_time = 1000.0 * m_format.m_frames / (m_sink_frameSize * m_format.m_sampleRate);
+            sleep_time = 1000.0 * m_format.m_frames / m_format.m_sampleRate;
```

Comment:
Sorry but I not understand this change (I'm not saying is wrong, only if you can explain its motivation).

Was wrong before for all audio formats: AC3, DTS, etc.  
Why not causing side effects?

Reply:
The original code was wrong,  but it was only ran when writing to the sink failed. Therefore it was never seen. The division with m_sink_frameSize is wrong, as the m_frames already relate to sampleRate.

Reply:
It's really good news if this was wrong and now it's been found. 

Now i just wonder if use ratio here makes sense?  🙂

`sleep_time = 1000.0 * m_format.m_frames / m_format.m_sampleRate / ratio;`

Reply:
No - that makes no sense, I tell you why: The sink works in periods and samplerate. It does not care at all about the payload that is inside the frames. Here we really wanted to wait the time the "packet" is with 100% guarantee processed by the sink. And the sink still needs frame / sampleRate processing time, even for the TrueHD case where only (frame /2) / sampleRate payload is integrated.

Which is also again the reason that I really want to fix the TrueHD packaging so that the payload size is again the "transport" size. Else the sink has to forward 3840 frames, while the payload is only 1920 frames - this is entirely not good.